src/Xrefcheck/Core.hs

```diff
@@ -77,8 +74,57 @@ data Reference = Reference
   , rAnchor :: Maybe Text
     -- ^ Section or custom anchor tag.
   , rPos    :: Position
+    -- ^ Position in source file.
+  , rInfo   :: ReferenceInfo
+    -- ^ More info about the link.
   } deriving stock (Show, Generic)
 
+-- | Info about the reference.
+data ReferenceInfo
+  = RIExternal
+    -- ^ Reference to a file at outer site, e.g @[d](http://www.google.com/doodles)@
+  | RIOtherProtocol
+    -- ^ Entry not to be processed, e.g. @mailto:e-mail@
+  | RIFileLocal
+    -- ^ Reference to this file, e.g. @[a](#header)@
+  | RIFileAbsolute
+    -- ^ Reference to a file absolute to the root, e.g. @[c](/folder/file#header)@
+  | RIFileRelative
+    -- ^ Reference to a file relative to given one, e.g. @[b](folder/file#header)@
+  deriving stock (Show, Generic)
+
+-- | Compute the 'ReferenceInfo' corresponding to a given link.
+referenceInfo :: Text -> ReferenceInfo
+referenceInfo link = case toString link of
+  [] -> RIFileLocal
+  c : _             | isPathSeparator c -> RIFileAbsolute
+  '.' : c : _       | isPathSeparator c -> RIFileRelative
+  '.' : '.' : c : _ | isPathSeparator c -> RIFileRelative
```

Comment:
Don't you like `PathSep` pattern?

Reply:
As I have been trying different approaches during the refactor, at some point I removed it but now it maybe useful again. I will try to add it and see the result!